composeApp/src/androidMain/kotlin/org/hh/museli/MainActivity.kt

```diff
@@ -37,13 +38,11 @@ class MainActivity : ComponentActivity() {
             requestPermissionLauncher.launch(READ_MEDIA_AUDIO)
         }
 
-        val libraryProvider: suspend () -> Library? = {
+        val libraryProvider: suspend () -> Library = {
             val dir = FileKit.pickDirectory()?.path
-            if (dir != null) {
-                AndroidLibrary(dir, applicationContext)
-            } else {
-                null
-            }
+            check(dir != null, ::nullError)
+            check(dir.startsWith("/tree/primary:")) { "only directories on primary storage are supported, invalid dir: $dir" }
```

Comment:
TLDR: Because android is a bitch.....

In newer SDK, we don't get the absolute path of the file but get this weird uri thing where the part before `:` denotes what storage device we are accessing, and the part after it is the file we are accessing.
`/tree/primary:` is the URI prefix for media present on primary storage of the device. I don't know what external storage devices are labelled as because I couldn't find any documentation on that and I didn't bother to test and find out. Now the music player API that we are using isn't compatible with this URI style path, it wants absolute path, so I translate URI path into absolute path using `applicationContext.contentResolver.query`. I think there are ways to avoid this mess, but I'll clean that up in another PR